backendutil/transform.go

```diff
@@ -0,0 +1,96 @@
+package backendutil
+
+import (
+	"io"
+
+	"github.com/emersion/go-smtp"
+)
+
+// TransformBackend is a backend that transforms messages.
+type TransformBackend struct {
+	Backend smtp.Backend
+
+	Transform          func(session *smtp.Session, username string) TransformHandler
+	AnonymousTransform func(session *smtp.Session) TransformHandler
+}
+
+// TransformHandler is a container for transforming funcs.
+type TransformHandler interface {
```

Comment:
This requires users to create an object that implements TransformHandler, right? Isn't it the same as implementing the actual Session interface? I guess you wanted to have a structure with func(..) variables here.

Reply:
Thanks. You are right, I probably wanted a struct instead of an interface. Will make the change later today.